rs/execution_environment/src/scheduler.rs

```diff
@@ -824,19 +824,14 @@ impl SchedulerImpl {
                 .metrics
                 .round_inner_heartbeat_overhead_duration
                 .start_timer();
-            // Remove all remaining `Heartbeat`, `GlobalTimer`, and `OnLowWasmMemory` tasks
+            // Remove all remaining `Heartbeat`, `GlobalTimer` tasks
             // because they will be added again in the next round.
             for canister_id in &heartbeat_and_timer_canister_ids {
                 let canister = state.canister_state_mut(canister_id).unwrap();
-                canister.system_state.task_queue.retain(|task| match task {
-                    ExecutionTask::Heartbeat
-                    | ExecutionTask::GlobalTimer
-                    | ExecutionTask::OnLowWasmMemory => false,
-                    ExecutionTask::PausedExecution { .. }
-                    | ExecutionTask::PausedInstallCode(..)
-                    | ExecutionTask::AbortedExecution { .. }
-                    | ExecutionTask::AbortedInstallCode { .. } => true,
-                });
+                canister
+                    .system_state
+                    .task_queue
+                    .remove_heartbeat_and_global_timer();
```

Comment:
In `remove_heartbeat_and_global_timer` we will not delete `ExecutionTask::OnLowWasmMemory` as previously the case. But since `ExecutionTask::OnLowWasmMemory` has never been added to the task queue, that will not create the problem. The initial ideal was to add it to the task queue as all other tasks, but after creating the new struct we do not need this anymore. This will also close https://dfinity.atlassian.net/browse/EXC-1666.